src/containers/pages/IRS/assignments/index.tsx

```diff
@@ -79,19 +83,19 @@ const IRSAssignmentPlansList = (props: PlanAssignmentsListProps) => {
           }
         })
         .catch(err => {
-          // console.log('ERR', err)
+          growl(err.message, { type: toast.TYPE.ERROR });
         });
-      setLoading(false);
     } catch (e) {
-      // do something with the error?
+      growl(e.message, { type: toast.TYPE.ERROR });
     }
   }
 
   useEffect(() => {
     loadData();
+    return () => controller.abort();
   }, []);
 
-  if (loading) {
+  if (plans.length < 1) {
     return <Loading />;
   }
```

Comment:
Feels like this should be an if-else-if block so we don't render the spinner when we're done waiting for the response.
```
if (loading) {
  return <Loading />
} else if (plans.length < 1) {
  return <p>Sorry, no plans were found</p>
}
```
(without magical strings of course!)

Reply:
I agree that would be ideal, however for this component, loading is a "computed variable" of `plans.length`.

Reply:
#798